lua/rustaceanvim/lsp/init.lua

```diff
@@ -320,32 +323,19 @@ M.reload_settings = function(bufnr)
   return clients
 end
 
----Updates the target architecture setting for the LSP client associated with the given buffer.
----@param bufnr? number The buffer number, defaults to the current buffer
----@param target? string The target architecture. Defaults to nil(the current buffer's target if not provided).
-M.set_target_arch = function(bufnr, target)
+---Updates LSP client target architecture setting.
+---@param exclude_rustc_target? string|nil Cargo target triple (e.g., 'x86_64-unknown-linux-gnu') to filter rust-analyzer clients
+M.set_target_arch = function(exclude_rustc_target)
   ---@param client vim.lsp.Client
-  restart(bufnr, function(client)
-    -- Get current user's rust-analyzer target
-    local current_target = vim.tbl_get(client, 'config', 'settings', 'rust-analyzer', 'cargo', 'target')
-
-    if not target then
-      if not current_target then
-        vim.notify('Using default OS target architecture.', vim.log.levels.INFO)
-      else
-        vim.notify('Target architecture is already set to the default OS target.', vim.log.levels.INFO)
-      end
-      return
-    end
-
+  restart(exclude_rustc_target, function(client)
     with_rustc_target_architectures(function(rustc_targets)
-      if target == nil or rustc_targets[target] then
-        client.settings['rust-analyzer'].cargo.target = target
-        client.notify('workspace/didChangeConfiguration', { settings = client.config.settings })
-        vim.notify('Target architecture updated successfully to: ' .. target, vim.log.levels.INFO)
+      if rustc_targets[exclude_rustc_target] then
+        client.settings['rust-analyzer'].cargo.target = exclude_rustc_target
+        client.notify('workspace/didchangeconfiguration', { settings = client.config.settings })
```

Comment:
issue: it appears a lot of code was accidentally lowercased (see also the log levels in this file).

```suggestion
        client.notify('workspace/didChangeConfiguration', { settings = client.config.settings })
```